### PR TITLE
nixos/munin: fix two issues with extraAutoPlugins

### PR DESCRIPTION
**Pull request by ToxicFrog**
## Description of changes

If extraAutoPlugins contains values that carry context (e.g. it comes from a flake input), the keys generated from them using baseNameOf inherit that context and the config doesn't compile.

This doesn't actually need to be an attrset anyways, so a bit of internal refactoring lets us fix this without changing the visible API.

It also tweaks the `sed` command in `internAndFixPlugins` to not mangle nix store paths, so it doesn't break plugins that are coming from nix tools like `writeShellScriptBin` rather than from munin-contrib.

Relevant: #17895 -- this makes it easier to use munin-contrib plugins from flake inputs or from the store. (Contrib still isn't packaged though.)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/monitoring/munin.nix [hunk 1]
```diff
@@ -83,14 +83,15 @@ let
...
     "cp -a '${path}' '${name}'";
 
   # Copy an entire tree of Munin plugins into a single directory in the Nix
-  # store, with no renaming.
-  # This is suitable for use with munin-node-configure --suggest, i.e.
-  # munin.extraAutoPlugins.
-  internManyPlugins = name: path:
+  # store, with no renaming. The output is suitable for use with
+  # munin-node-configure --suggest, i.e. munin.extraAutoPlugins.
+  # Note that this flattens the input; this is intentional, as node-configure
```
Comment (bjornfor) [inline nixos/modules/services/monitoring/munin.nix:92]: node-configure -> munin-node-configure?

Comment (ToxicFrog) [reply nixos/modules/services/monitoring/munin.nix]: Whoops, done. Thanks.

File: nixos/modules/services/monitoring/munin.nix [hunk 2]
```diff
@@ -375,7 +372,7 @@ in
 
     # Munin is hardcoded to use DejaVu Mono and the graphs come out wrong if
     # it's not available.
-    fonts.packages = [ pkgs.dejavu_fonts ];
+    fonts.fonts = [ pkgs.dejavu_fonts ];  # fonts.packages in unstable, but fonts.fonts in 2305
```
Comment (bjornfor) [inline nixos/modules/services/monitoring/munin.nix:375]: Unrelated change. Please put in a separate commit. (Same PR is fine.)

Comment (ToxicFrog) [reply nixos/modules/services/monitoring/munin.nix:375]: This is what I get for using `git add` without `-p` late at night -- that change shouldn't be merged, it's a local change to let my (nixos-23.03) systems load this (nixos-unstable) module.

I'll revert that, and I also found another munin-module bug related to `internAndFixPlugins` that I want to get a fix for into this branch later this afternoon.

File: nixos/modules/services/monitoring/munin.nix [hunk 3]
```diff
@@ -96,14 +96,22 @@ let
...
   internAndFixPlugins = name: intern-fn: paths:
     pkgs.runCommand name {} ''
       mkdir -p "$out"
       cd "$out"
       ${lib.concatStringsSep "\n" (map intern-fn paths)}
       chmod -R u+w .
-      find . -type f -exec sed -E -i '
-        s,(/usr)?/s?bin/,/run/current-system/sw/bin/,g
+      ${pkgs.findutils}/bin/find . -type f -exec ${pkgs.gnused}/bin/sed -E -i '
+        \%/nix/store/%! s,(/usr)?/s?bin/,/run/current-system/sw/bin/,g
```
Comment (bjornfor) [inline nixos/modules/services/monitoring/munin.nix:114]: Use $NIX_STORE (?) env var instead of hardcoding the path?

Comment (ToxicFrog) [reply nixos/modules/services/monitoring/munin.nix]: This sent me down a bit of a rabbit hole, because it's `$NIX_STORE_PATH` when *invoking* nix (and not guaranteed to be set), but `$NIX_PATH` once the builder actually runs (and is guaranteed). Fixed, thanks.

**Review by bjornfor — APPROVED** [review]

File: nixos/modules/services/monitoring/munin.nix [hunk 4]
```diff
@@ -96,15 +96,23 @@ let
 
   # Use the appropriate intern-fn to copy the plugins into the store and patch
   # them afterwards in an attempt to get them to run on NixOS.
+  # This is a bit hairy because we can't just fix shebangs; lots of munin plugins
+  # hardcode paths like /sbin/mount rather than truing $PATH, so we have to look
```
Comment (bjornfor) [inline nixos/modules/services/monitoring/munin.nix:100]: truing -> trusting?

Comment (ToxicFrog) [reply nixos/modules/services/monitoring/munin.nix]: Done.

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Thanks!
